fastlane_core/lib/fastlane_core/project.rb

```diff
@@ -301,13 +312,33 @@ def self.xcode_list_timeout
     end
 
     # @internal to module
-    # runs the specified command and kills it if timeouts
-    # @raises Timeout::Error if timeout is passed
+    def self.xcode_build_settings_timeout
+      (ENV['FASTLANE_XCODEBUILD_SETTINGS_TIMEOUT'] || 10).to_i
+    end
+
+    # @internal to module
+    def self.xcode_build_settings_retries
+      (ENV['FASTLANE_XCODEBUILD_SETTINGS_RETRIES'] || 3).to_i
+    end
+
+    # @internal to module
+    # runs the specified command and kills it if timeouts, optionally retries before killing
+    # @raises Timeout::Error if timeout is passed after all retry attempts
     # @returns the output
-    # Note: currently affected by fastlane/fastlane_core#102
-    def self.run_command(command, timeout: 0)
+    # Note: - currently affected by fastlane/fastlane_core#102
+    #       - retry feature added to solve fastlane#4059
+    def self.run_command(command, timeout: 0, retries: 0)
       require 'timeout'
-      @raw = Timeout.timeout(timeout) { `#{command}`.to_s }
+
+      tries = 1
+      begin
+        return Timeout.timeout(timeout) { `#{command}`.to_s }
```

Comment:
Yes I noticed, I can't say I know enough Ruby to understand it. Feels totally "wat?". 

Everything I read said to just not use `Timeout` (links in older comments here) but that means not using backticks. Actually not using `Timeout` was my first implementation, which was based on #4492. I changed it just because the core team didn't seem to agree with that approach.